packages/solana/src/client.ts

```diff
@@ -332,6 +339,11 @@ export class Web3Modal extends Web3ModalScaffold {
 
   public async switchNetwork(caipNetwork: CaipNetwork) {
     const caipChainId = caipNetwork.id
+
+    if (this.provider instanceof AuthProvider) {
+      await this.provider.switchNetwork(caipChainId)
+    }
```

Comment:
is this required? can't we init the provider with a network directly?

Reply:
but what if the user changes the network? the secure site will remain with old network